app/views/unattended/provisioning_templates/finish/windows_default_finish.erb

```diff
@@ -44,89 +45,104 @@ description: |
       <% end %>
     <% end %>
   <% end %>
 
+  <%#
+  Adjust power setting to 'High performance'
+  -%>
+  powershell /c "powercfg /setactive 8c5e7fda-e8bf-4a96-9a85-a6e23a8c635c"
+
   <% if host_param('localAdminAccountDisabled') %>
     echo Disabling %tempAdminUser%
     net user %tempAdminUser% %tempAdminUser% /active:no
   <% end %>
 
-  <% if host_param('ansible_port') == 5985 or host_param('ansible_winrm_scheme') == 'http' %>
-    cmd /c winrm set winrm/config/service @{AllowUnencrypted="true"}
+  <% if host_param('http-proxy').present? -%>
+    cmd /C "netsh winhttp set proxy <%= host_param('http-proxy') %>:<%= host_param('http-proxy-port') %>"
+  <% end %>
+
+  <% unless host_param('computerDomain') %>
+    powershell /c "Get-NetConnectionProfile -InterfaceAlias \"Ethernet0\" | Set-NetConnectionProfile -NetworkCategory <%= network_location %>"
   <% end %>
 
-  <% if host_param('ansible_winrm_transport') == 'basic' %>
+  <% if host_param('ansible_user').present? %>
+    <% if host_param('create_ansible_user') %>
+    powershell /c "set-localuser -name <%= host_param('ansible_user') %> -passwordneverexpires 1"
+    <% end %>
+    powershell /c "Enable-PSRemoting"
+    cmd /c "netsh advfirewall firewall add rule name="WinRM-HTTP" dir=in localport=5985 protocol=TCP action=allow"
+    cmd /c winrm set winrm/config/service @{AllowUnencrypted="true"}
     cmd /c winrm set winrm/config/client/auth @{Basic="true"}
     cmd /c winrm set winrm/config/service/auth @{Basic="true"}
   <% end %>
 
-  <% if host_param('ansible_winrm_transport') == 'credssp' %>
-    cmd /c winrm set winrm/config/client/auth @{CredSSP="true"}
-    cmd /c winrm set winrm/config/service/auth @{CredSSP="true"}
+  <% if host_param('ping') %>
+    cmd /c "netsh advfirewall firewall add rule name=\"Enable IPv4 ICMP\" dir=in protocol=icmpv4 action=allow"
   <% end %>
 
-  <% if host_param('ansible_winrm_transport') == 'certificate' %>
-    cmd /c winrm set winrm/config/client/auth @{Certificate="true"}
-    cmd /c winrm set winrm/config/service/auth @{Certificate="true"}
+  <% if host_param('remote_desktop') %>
+    cmd /c "netsh advfirewall firewall set rule group=\"remote desktop\" new enable=Yes"
+    cmd /c "netsh advfirewall firewall set rule group=\"remotedesktop\" new enable=Yes"
   <% end %>
 
-  <%= snippet 'Windows network' %>
-
-  <% if foreman_url('user_data') %>
-    echo execute user data script
-    IF EXIST c:\deploy\user_data.ps1 powershell.exe -OutputFormat text -command c:\deploy\user_data.ps1
-  <% end -%>
-
   <% if puppet_enabled %>
-    echo Installing puppet
+    echo Downloading Puppet installer
+    wget "<%= host_param('win_puppet_source') %>" -O C:\puppet-agent-x64-latest.msi
+    echo Configuring Puppet
+    mkdir "C:\ProgramData\PuppetLabs\puppet\etc"
+    echo [main] > "C:\ProgramData\PuppetLabs\puppet\etc\puppet.conf"
```

Comment:
Why is this needed now? I suspected the `puppet.msi` (which is still there) would be sufficient.